core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -151,20 +151,21 @@ public InternalEngine(EngineConfig engineConfig, boolean skipInitialTranslogReco
                     throw e;
                 }
             }
+
             this.translog = translog;
             manager = createSearcherManager();
             this.searcherManager = manager;
             this.versionMap.setManager(searcherManager);
             try {
-                if (skipInitialTranslogRecovery) {
+                if (openMode == EngineConfig.OpenMode.OPEN_INDEX_CREATE_TRANSLOG) {
```

Comment:
this is now redundant. openTranslog commits when openMode == EngineConfig.OpenMode.OPEN_INDEX_CREATE_TRANSLOG